src/app/config/config.c

```diff
@@ -3010,9 +3031,16 @@ is_local_addr, (const tor_addr_t *addr))
      * it was on net 0, which is already done by tor_addr_is_internal.
      */
     if ((last_resolved_addr & (uint32_t)0xffffff00ul)
-        == (ip & (uint32_t)0xffffff00ul))
+        == (addr_ipv4 & (uint32_t)0xffffff00ul))
       return 1;
   }
+
+  /* Check for the same IPv6 /48 as the directory server */
+  if (tor_addr_family(other_addr) == AF_INET6) {
+    if (router_get_my_routerinfo() && tor_addr_is_valid(addr_ipv6, 0))
```

Comment:
If is_local_addr() has already checked router_get_my_routerinfo(), you don't need to check it again here.